spec/datadog/appsec/processor_spec.rb

```diff
@@ -112,6 +112,26 @@
       it { expect(described_class.new.send(:load_ruleset)).to be true }
     end
 
+    context 'when ruleset is :strict' do
+      let(:ruleset) { :strict }
+
+      before do
+        expect(Datadog::AppSec::Assets).to receive(:waf_rules).with(:strict).and_call_original.twice
+      end
+
+      it { expect(described_class.new.send(:load_ruleset)).to be true }
+    end
+
+    context 'when ruleset is :risky' do
+      let(:ruleset) { :risky }
+
+      before do
+        expect(Datadog::AppSec::Assets).to receive(:waf_rules).with(:risky).and_call_original.twice
+      end
```

Comment:
Ah, it's because of the way Processor initialization + these specs are written.

1. `Processor.new` calls `#load_ruleset` internally (designed as such to have either a fully working Processor instance or an exception)
2. but we then `send(:load_ruleset)` in these examples to actually unit test that method in isolation, which is very artificial.

IRL the method is only called once.